bot.py

```diff
@@ -412,6 +429,20 @@ async def on_tree_error(
         raise error
 
 
+async def distribute_any_users(difficulty_users_dict, any_users):
+    # First, ensure all bins have an even number of users
```

Comment:
this doesn't seem right because the random distribution afterwards can just make them all odd again. I think you just want to upgrade size-1 bins up to size-2.